src/plugins/kernel_install/utils.sh

```diff
@@ -564,35 +575,48 @@ function kernel_uninstall()
   cmd_manager "$flag" "sudo mkdir -p $REMOTE_KW_DEPLOY"
   cmd_manager "$flag" "sudo touch '$INSTALLED_KERNELS_PATH'"
 
+  all_kernels=$(list_installed_kernels 'SILENT' 1 1 "$prefix")
+  IFS=, read -r -a available_kernels <<< "$all_kernels"
+  mapfile -t deduped_all_kernels < <(printf "%s\n" "${available_kernels[@]}" | sort -u)
+
+  kw_managed_kernels=$(list_installed_kernels 'SILENT' 1 '' "$prefix")
+  IFS=, read -r -a available_kernels <<< "$kw_managed_kernels"
+  mapfile -t deduped_kw_managed_kernels < <(printf "%s\n" "${available_kernels[@]}" | sort -u)
```

Comment:
This code is very similar to the code above it; create a function for that to avoid code duplication.